pkg/controller/database/database_controller_test.go

```diff
@@ -0,0 +1,198 @@
+package database
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/schemahero/schemahero/pkg/apis"
+	databasesv1alpha4 "github.com/schemahero/schemahero/pkg/apis/databases/v1alpha4"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+)
+
+// TestDbControllerTest uses envtest (see https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest#Environment
+// and https://github.com/kubernetes-sigs/controller-runtime/blob/master/FAQ.md#q-wheres-the-fake-client--how-do-i-use-it)
+// This starts a real kube-api and etcd and runs the controller against it. The tests then
+// become about submitting requests to the kube API and asserting on the state at the end
+//
+// You will need to run 'make envtest' to install the kube api/etcd binaries
+func TestDbControllerTest(t *testing.T) {
+	// create envtest instance, point to binaries
+	// and load CRDs
+	env := &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			filepath.Join("..", "..", "..", "config", "crds", "v1"),
+		},
+		BinaryAssetsDirectory: filepath.Join("/tmp", "kubebuilder", "bin"),
```

Comment:
How about `os.TempDir()` instead of `"/tmp"`?

Reply:
I was having some permissions issues with the os TMPDIR when installing binaries, but seems to be gone now (the upgrade to Big Sur might have done something). In any case fair point.